### PR TITLE
r.univar: add JSON support

### PR DESCRIPTION
**Pull request by kritibirda26**
Use parson to add json output format support to the r.univar module.

Expected JSON schema:

At the root is an array, each element of which has the stats for a given zone. The percentile option allows the user to request specific percentiles which are written for each zone object as `percentile_%d` keys.

```
[
    {
        "zone_number": <int>,
        "zone_category": <string>,
        "n": <int>,
        "null_cells": <int>,
        "cells": <int>,
        "min": <double>,
        "max": <double>,
        "range": <double>,
        "mean": <double>,
        "mean_of_abs": <double>,
        "stddev": <double>,
        "variance": <double>,
        "coeff_var": <double>,
        "sum": <double>,
        "first_quartile": <double>,
        "median": <double>,
        "third_quartile": <double>,
        "percentile_98": <double>
    }
]
```

### CODE REVIEW

**Comment by wenzeslaus** [comment]
Can you please compare this schema to what db.univar generates? Maybe they don't need to be the same, but if they are not, we need to have a good reason for that.

**Comment by echoix** [comment]
> @wenzeslaus Can you please point me to the db.univar module, I am unable to find it.

https://github.com/OSGeo/grass/tree/f13b37931af919be04fa14f392a1f9ea9aa183fd/scripts/db.univar

**Review by cwhite911 — CHANGES_REQUESTED** [review]
Update json output and tests to match `v.univar`.

**Comment by kritibirda26** [comment]
@cwhite911 updated.

File: raster/r.univar/stats.c [hunk 1]
```diff
@@ -244,17 +292,57 @@ int print_stats(univar_stat *stats)
...
+                if (format == JSON) {
+                    percentiles_array_value = json_value_init_array();
+                    percentiles_array = json_array(percentiles_array_value);
+                }
+
                 for (i = 0; i < stats[z].n_perc; i++) {
-                    char buf[24];
+                    char buf[24], buf2[36];
 
                     sprintf(buf, "%.15g", stats[z].perc[i]);
```
Comment (nilason) [inline raster/r.univar/stats.c:323]: Even if not introduced by this PR, you might change `sprintf` to the more safer `snprintf`:
```suggestion
                    snprintf(buf, sizeof(buf), "%.15g", stats[z].perc[i]);
```

**Review by nilason — CHANGES_REQUESTED** [review]

File: raster/r.univar/stats.c [hunk 2]
```diff
@@ -244,17 +292,57 @@ int print_stats(univar_stat *stats)
...
+                JSON_Object *percentile_object;
+
+                if (format == JSON) {
+                    percentiles_array_value = json_value_init_array();
+                    percentiles_array = json_array(percentiles_array_value);
+                }
+
                 for (i = 0; i < stats[z].n_perc; i++) {
-                    char buf[24];
+                    char buf[24], buf2[36];
```
[the remark this answers is not in the document]
Comment (nilason) [reply raster/r.univar/stats.c:321]: This causes the CI runners to fail.

**Review by nilason — APPROVED** [review]

**Review by cwhite911 — APPROVED** [review]
@kritibirda26 nice work!

**Review by nilason — APPROVED** [review]
Code looks good to me, I haven't tested though.

**Pull request merged**